Re-adding draft doc for archival store and service (#23278)
MystenLabs/sui: docs/content/guides/developer/getting-started/data-serving.mdx

| -------- | ------- | ------- |
| :heavy_check_mark: April 2025 | Beta release of initial set of polling-based APIs. | You can start validating the initial gRPC integration from your application and share feedback on the improvements you want to see. |
| :heavy_check_mark: July 2025 | Beta release of streaming APIs and the remaining set of polling-based APIs. | If your use case requires streaming low-latency data, this is an apt time to start validating that integration. Also, the functionality of the API coverage will be complete at this point, so you can start migrating your application in non-production environments. |

> | September-October 2025 | GA release of polling-based and streaming APIs. | You can start migration and cutover of your application in the production environment. **JSON-RPC will be deprecated at this point and migration notice period will start.** |

Reviewer: ```suggestion
| September-October 2025 | GA release of polling-based and streaming APIs. | Begin migration and cutover of your application in the production environment. **JSON-RPC will be deprecated at this point and migration notice period will start.** |
```

| December 2025 | GA release of GraphQL RPC Server and Indexer 2.0. | You can start migration and cutover of your application in the production environment. |
| April 2026 | End of migration timeline. | **JSON-RPC will be fully deactivated at this point.** This timeline assumes about 7 months of migration notice period. |

### Archival Store and Service

As Full nodes enforce limited retention for scalability, long-term access to historical on-chain data becomes essential for developers. The [Archival Store and Service](../../../concepts/archival-store.mdx) enable access to this data through a pluggable storage backend (like [Bigtable](https://cloud.google.com/bigtable)) and a [gRPC interface compatible with the `LedgerService`](../../../concepts/grpc-overview.mdx).

> This infrastructure serves as the historical backbone for GraphQL RPC, gRPC-based apps, and data platforms, providing efficient point-lookups for old transactions, checkpoints, and object states, even after Full nodes have pruned them.

Reviewer: ```suggestion
This infrastructure serves as the historical backbone for GraphQL RPC, gRPC-based apps, and data platforms, providing efficient point-lookups for old transactions, checkpoints, and object states, even after full nodes have pruned them.
```
